forge-game/src/main/java/forge/game/card/CardFactoryUtil.java

```diff
@@ -239,14 +239,14 @@ public static boolean handleHiddenAgenda(Player player, Card card) {
 
         card.turnFaceDown();
         card.addMayLookAt(player.getGame().getNextTimestamp(), ImmutableList.of(player));
-        card.addSpellAbility(abilityRevealHiddenAgenda(card));
+        ki.addSpellAbility(abilityRevealHiddenAgenda(card));
```

Comment:
this somehow breaks it

Reply:
shouldn't the Keyword still add the SpellAbility to the Card?

Or do I need Extra check for TurnUp somewhere else?

Reply:
should be fixed, i used Restriction instead of Condition because i somehow could turned it back face-down when i shouldn't